lib/std/hash_map.zig

```diff
@@ -818,26 +818,54 @@ pub fn HashMapUnmanaged(
             hm: *const Self,
             index: Size = 0,
 
+            const used_mask = @bitCast(u8, Metadata{
+                .used = 1,
+                .fingerprint = 0,
+            });
             pub fn next(it: *Iterator) ?Entry {
                 assert(it.index <= it.hm.capacity());
                 if (it.hm.size == 0) return null;
 
                 const cap = it.hm.capacity();
-                const end = it.hm.metadata.? + cap;
                 var metadata = it.hm.metadata.? + it.index;
-
-                while (metadata != end) : ({
-                    metadata += 1;
-                    it.index += 1;
-                }) {
-                    if (metadata[0].isUsed()) {
-                        const key = &it.hm.keys()[it.index];
-                        const value = &it.hm.values()[it.index];
+                if (comptime builtin.zig_backend != .stage2_c and // TODO
+                    // https://github.com/ziglang/zig/issues/13782
+                    (builtin.zig_backend != .stage2_llvm or builtin.cpu.arch.endian() != .Big))
+                {
+                    comptime var vec_size: u8 = 16;
```

Comment:
I think this should probably be using `std.simd.suggestVectorSize` to optimize further when for example AVX or newer is targeted.